src/main/common/interest.ts

```diff
@@ -19,7 +19,8 @@ export async function getInterestDetails (claim: Claim): Promise<object> {
   }
 
   const todayDate: Moment = MomentFactory.currentDate()
-  const noOfDays: number = todayDate.diff(interestDate, 'days') + 1
+  const noOfDays: number = interestDate.diff(interestDate, 'days') === 0 ?
```

Comment:
That makes no sense to me. Doing diff on the same date instance (`interestDate.diff(interestDate, 'days')`) will always give 0 e.g. difference between today and today is always 0.

Another thing is that if diff always evaluates to 0 then `todayDate.diff(interestDate, 'days')` is always executed which basically reverts the calculation logic to the initial version which was clarifies as not woking properly.